elixir/apps/api/lib/api/client/channel.ex

```diff
@@ -153,261 +122,289 @@ defmodule API.Client.Channel do
+      )
+      when old_config != config do
+    socket = assign(socket, client: %{socket.assigns.client | account: account})
 
     :ok =
       push(socket, "config_changed", %{
-        interface:
-          Views.Interface.render(%{
-            socket.assigns.client
-            | account: account
-          })
+        interface: Views.Interface.render(socket.assigns.client)
       })
 
     {:noreply, socket}
   end
 
-  # Message is scheduled by schedule_expiration/1 on topic join to be sent
-  # when the client token/subject expires
-  def handle_info(:token_expired, socket) do
-    OpenTelemetry.Ctx.attach(socket.assigns.opentelemetry_ctx)
-    OpenTelemetry.Tracer.set_current_span(socket.assigns.opentelemetry_span_ctx)
-
-    OpenTelemetry.Tracer.with_span "client.token_expired" do
-      push(socket, "disconnect", %{reason: :token_expired})
-      {:stop, {:shutdown, :token_expired}, socket}
-    end
+  def handle_info({:deleted, %Accounts.Account{}}, socket) do
+    disconnect(socket)
   end
 
-  # This event is broadcasted when client was changed (eg. renamed, verified, etc.)
-  def handle_info({:updated, %Clients.Client{} = updated_client}, socket) do
-    OpenTelemetry.Ctx.attach(socket.assigns.opentelemetry_ctx)
-    OpenTelemetry.Tracer.set_current_span(socket.assigns.opentelemetry_span_ctx)
-
-    OpenTelemetry.Tracer.with_span "client.updated" do
-      client = socket.assigns.client
-      socket = assign(socket, client: updated_client)
+  # ACTOR_GROUP_MEMBERSHIPS
 
-      socket =
-        if client.verified_at != updated_client.verified_at do
-          # Re-initialize since list of allowed_resources could have changed
-          {:ok, socket} = init(socket)
+  def handle_info(
+        {:created, %Actors.Membership{actor_id: actor_id, group_id: group_id}},
+        %{assigns: %{client: %{actor: %{id: id}}}} = socket
+      )
+      when id == actor_id do
+    # 1. Get existing resources
+    existing_resources = get_resources(socket)
+
+    # 2. Update our state
+    new_policies =
+      Policies.all_policies_for_actor_group_id!(socket.assigns.client.account_id, group_id)
+      |> Enum.map(fn policy -> {policy.id, policy} end)
+      |> Map.new()
+
+    socket =
+      socket
+      |> assign(policies: Map.merge(socket.assigns.policies, new_policies))
+      |> assign(group_ids: MapSet.put(socket.assigns.group_ids, group_id))
+
+    # 3. Get new resources
+    new_resources = get_resources(socket) -- existing_resources
+
+    # 4. Push new resources to the client
+    for resource <- new_resources do
+      push(socket, "resource_created_or_updated", Views.Resource.render(resource))
+    end
 
-          socket
-        else
-          socket
-        end
+    {:noreply, socket}
+  end
 
-      {:noreply, socket}
+  def handle_info(
+        {:deleted, %Actors.Membership{actor_id: actor_id, group_id: group_id}},
+        %{assigns: %{client: %{actor: %{id: id}}}} = socket
+      )
+      when id == actor_id do
```

Comment:
Can you do this as part of pattern matching on the message?

Early return would also reduce the right-wards drift.